libflux/src/semantic/types.rs

```diff
@@ -645,24 +716,47 @@ impl fmt::Display for Function {
     }
 }
 
-impl Substitutable for HashMap<String, MonoType> {
+impl<T: Substitutable> Substitutable for HashMap<String, T> {
```

Comment:
This kind of abstraction in Rust still amazes me. It reads well too.

Reply:
I love it. It makes implementing this trait so composable.